psamm/datasource/sbml.py

```diff
@@ -709,11 +709,51 @@ def _add_flux_bounds(self, r_id, model, flux_limits, equation):
             upper_str = text_type(flux_limits[r_id][1])
         return lower_str, upper_str
 
+    def _add_gene_associations(self, r_id, r_genes, gene_set, r_tag):
+        if r_id in r_genes:
+            genes = ET.SubElement(
+                r_tag, _tag('geneProductAssociation', FBC_V2))
+            if isinstance(r_genes[r_id], list):
+                e = Expression(
+                    And(*(Variable(i) for i in r_genes[r_id])))
+            else:
+                e = Expression(r_genes[r_id])
+            gene_stack = [(e._root, genes)]
+            while len(gene_stack) > 0:
+                current, parent = gene_stack.pop()
+                if isinstance(current, Or):
+                    gene_tag = ET.SubElement(parent, _tag('or', FBC_V2))
+                elif isinstance(current, And):
+                    gene_tag = ET.SubElement(parent, _tag('and', FBC_V2))
+                elif isinstance(current, Variable):
+                    gene_tag = ET.SubElement(parent, _tag(
+                        'geneProductRef', FBC_V2))
+                    gene_tag.set(_tag('geneProduct', FBC_V2),
+                                 current.symbol)
+                    if current.symbol not in gene_set:
+                        gene_set.add(current.symbol)
+                if isinstance(current, (Or, And)):
+                    for item in current:
+                        gene_stack.append((item, gene_tag))
+
+    def _add_gene_list(self, parent_tag, all_genes):
+        """ list all gene products"""
+        list_all_genes = ET.SubElement(parent_tag, _tag(
+            'listOfGeneProducts', FBC_V2))
+        gene_ids = {}
+        for gene in sorted(all_genes):
+            gene_tag = ET.SubElement(list_all_genes, _tag('gene', FBC_V2))
+            id = self._create_unique_id(gene_ids, self._make_safe_id(gene))
+            gene_ids[gene] = id
+            gene_tag.set(_tag('id', FBC_V2), 'g_' + id)
+            gene_tag.set(_tag('label', FBC_V2), gene)
+
     def write_model(self, file, model):
         """Write a given model to file"""
 
         ET.register_namespace('mathml', MATHML_NS)
         ET.register_namespace('xhtml', XHTML_NS)
+        ET.register_namespace('fbc', FBC_V2)
```

Comment:
This has to go in the first commit. Either a separate commit that only adds this namespace or the first of your commits that uses the namespace.